intake_stac/tests/test_catalog.py

```diff
@@ -53,180 +62,199 @@ def intake_stac_cat():
-
-
-def test_asset_describe(pystac_item):
-    item = StacItem(pystac_item)
-    key = 'B1'
-    asset = item[key]
-    d = asset.describe()
-
-    assert d['name'] == key
-    assert d['container'] == 'xarray'
-    assert d['plugin'] == ['rasterio']
-    assert d['args']['urlpath'] == asset.urlpath
-    assert d['description'] == asset.description
-    # NOTE: note sure why asset.metadata has 'catalog_dir' key ?
-    # assert d['metadata'] == asset.metadata
-
-
-def test_asset_missing_type(pystac_item):
-    key = 'B1'
-    asset = pystac_item.assets.get('B1')
-    asset.media_type = ''
-    with pytest.warns(Warning, match='STAC Asset'):
+class TestCatalog:
+    def test_init_catalog_from_url(self):
+        cat = StacCatalog(cat_url)
+        assert isinstance(cat, intake.catalog.Catalog)
+        assert cat.name == 'test'
+        assert cat.discover()['container'] == 'catalog'
+        assert int(cat.metadata['stac_version'][0]) >= 1
+
+        cat = StacCatalog.from_url(cat_url)
+        assert isinstance(cat, intake.catalog.Catalog)
+        assert cat.name == 'test'
+        assert cat.discover()['container'] == 'catalog'
+        assert int(cat.metadata['stac_version'][0]) >= 1
+
+        # test kwargs are passed through
+        cat = StacCatalog.from_url(cat_url, name='intake-stac-test')
+        assert 'intake-stac-test' == cat.name
+
+    def test_init_catalog_from_pystac_obj(self, pystac_cat):
+        cat = StacCatalog(pystac_cat)
+        assert isinstance(cat, intake.catalog.Catalog)
+        assert cat.discover()['container'] == 'catalog'
+        assert cat.name == 'test'
+        assert cat.name == pystac_cat.id
+
+        # test kwargs are passed through
+        cat = StacCatalog(pystac_cat, name='intake-stac-test')
+        assert 'intake-stac-test' == cat.name
+
+    def test_init_catalog_with_wrong_type_raises(self, pystac_cat):
+        with pytest.raises(ValueError):
+            StacCollection(pystac_cat)
+
+    def test_init_catalog_with_bad_url_raises(self):
+        # json.decoder.JSONDecodeError or FileNotFoundError
+        with pytest.raises(Exception):
+            StacCatalog('https://raw.githubusercontent.com/')
+
+    def test_serialize(self, intake_stac_cat):
+        cat_str = intake_stac_cat.serialize()
+        assert isinstance(cat_str, str)
+
+    def test_cat_entries(self, intake_stac_cat):
+        assert list(intake_stac_cat)
+        assert all(
+            [isinstance(v, (LocalCatalogEntry, Catalog)) for _, v in intake_stac_cat.items()]
+        )
+
+    def test_cat_name_from_pystac_catalog_id(self, intake_stac_cat):
+        assert intake_stac_cat.name == 'test'
+
+
+class TestCollection:
+    def test_cat_from_collection(self, pystac_col):
+        cat = StacCollection(pystac_col)
+        subcat_name = 'landsat-8-l1'
+        assert cat.name == pystac_col.id
+        assert subcat_name in cat
+        # This is taking way too long
```

Comment:
I've disabled the rest of it test. It was taking a while. Is that expected?

Reply:
has the catalog changed? does it hang if the item id doesn't exist? it shouldn't take long by the look of things